File: java/findDisappearedNumbers.java
/*
 * @Author: Tungbo
 * @Date: 2020-10-14 14:27:54
 * @LastEditTime: 2021-07-26 15:10:51
 * @LastEditors: Tungbo
 * @Description: leecode: 
 */
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

class findDisappearedNumbers {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        HashMap<Integer,Boolean> temp = new HashMap<>();
        for (int i=0;i<nums.length;i++) {
            temp.put(nums[i],true);
        }
        List<Integer> result = new ArrayList<>();
        for (int i=1;i<=nums.length;i++) {
            if (!temp.containsKey(i))
                result.add(i);
        }
        return result;
    }
}

File: kotlin/findDisappearedNumbers.kt
/*
 * @Author: Tungbo
 * @Date: 2020-10-14 15:05:18
 * @LastEditTime: 2021-06-07 11:26:46
 * @LastEditors: Tungbo
 * @Description: leecode: 
 */
class findDisappearedNumbers {
    fun findDisappearedNumbers(nums: IntArray): List<Int> {
        for (index in nums.indices) {
            val newIndex = Math.abs(nums[index]) - 1;
            if(nums[newIndex]>0)
                nums[newIndex] *= -1
        }
        val result : MutableList<Int> =ArrayList()
        for ((index,e) in nums.withIndex()) {
            if (e>0) {
                result.add(index+1)
            }
        }
        return result
    }
}